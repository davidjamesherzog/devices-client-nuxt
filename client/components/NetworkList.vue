<template>
  <div>
    <network-list-item
      v-for="network in networks"
      :key="network.id"
      :network="network"
    />
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator';
import Network from '~/client/models/Network';
import NetworkListItem from '~/components/NetworkListItem.vue';

@Component({
  name: 'NetworkList',
  components: {
    NetworkListItem
  }
})
export default class NetworkList extends Vue {
  @Prop({ type: Array })
  private networks!: Array<Network>;
}
</script>

<style lang="scss" scoped></style>
