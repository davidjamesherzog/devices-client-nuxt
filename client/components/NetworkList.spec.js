import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import NetworkList from '~/components/NetworkList.vue';

describe('NetworkList', () => {
  describe('template', () => {
    let localVue;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
    });

    test('should render network list', () => {
      const wrapper = mount(NetworkList, {
        localVue,
        propsData: {
          networks: [
            {
              id: 1,
              name: 'TestWifi',
              description: 'Test description',
              type: 'WIFI'
            }
          ]
        }
      });
      const id = wrapper.find('[data-list-item-id="1"]');
      expect(id).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
