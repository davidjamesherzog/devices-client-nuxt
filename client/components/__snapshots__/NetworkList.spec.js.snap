// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkList template should render network list 1`] = `
<div>
  <div data-list-item-id="1" class="box">
    <article class="media">
      <div class="media-left">
        <div class="content has-text-centered"><span class="icon has-text-primary is-large" data-card-icon=""><i class="mdi mdi-wifi mdi-48px"></i></span></div>
      </div>
      <div class="media-content">
        <div class="content">
          <p><strong data-list-item-title="1">TestWifi</strong> <small data-list-item-type="1">WIFI</small> <br> <span data-list-item-description="1">
            Test description
          </span></p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
